include/cuco/detail/static_multimap/device_view_impl.inl

```diff
@@ -502,19 +503,23 @@ class static_multimap<Key, Value, Scope, Allocator, ProbeSequence>::device_view_
 #if defined(CUCO_HAS_CUDA_BARRIER)
       cooperative_groups::memcpy_async(
         g,
-        output_begin + offset,
+        &thrust::raw_reference_cast(*(output_begin + offset)),
         output_buffer,
         cuda::aligned_size_t<alignof(value_type)>(sizeof(value_type) * num_outputs));
 #else
-      cooperative_groups::memcpy_async(
-        g, output_begin + offset, output_buffer, sizeof(value_type) * num_outputs);
+      cooperative_groups::memcpy_async(g,
+                                       &thrust::raw_reference_cast(*(output_begin + offset)),
+                                       output_buffer,
+                                       sizeof(value_type) * num_outputs);
 #endif  // end CUCO_HAS_CUDA_BARRIER
       return;
 #endif  // end CUCO_HAS_CG_MEMCPY_ASYNC
     }
+#pragma nv_diag_suppress 128  // warning: unreachable
     for (auto index = lane_id; index < num_outputs; index += g.size()) {
       *(output_begin + offset + index) = output_buffer[index];
     }
+#pragma nv_diag_default 128
```

Comment:
```suggestion
#endif  // end CUCO_HAS_CUDA_BARRIER
#endif  // end CUCO_HAS_CG_MEMCPY_ASYNC
    }
    if constexpr (not thrust::is_contiguous_iterator_v<OutputIt>) {
      for (auto index = lane_id; index < num_outputs; index += g.size()) {
        *(output_begin + offset + index) = output_buffer[index];
      }
    }
```
Changing the logic a bit may fix the issue.

Reply:
lol, yeah I was just about to type that.

Reply:
Ah, wait, this way the function does nothing for the case `not defined(CUCO_CG_MEMCPY_ASYNC) and thrust::is_contiguous_iterator_v<OutputIt>`.

Reply:
dang... you are right

Reply:
```cpp
#if defined(CUCO_HAS_CG_MEMCPY_ASYNC)
    constexpr bool uses_memcpy_async = thrust::is_contiguous_iterator_v<OutputIt>;
#else
    constexpr bool uses_memcpy_async = false;
#endif

    if constexpr (uses_memcpy_async) {
#if defined(CUCO_HAS_CUDA_BARRIER)
      cooperative_groups::memcpy_async(
        g,
        &thrust::raw_reference_cast(*(output_begin + offset)),
        output_buffer,
        cuda::aligned_size_t<alignof(value_type)>(sizeof(value_type) * num_outputs));
#else
      cooperative_groups::memcpy_async(g,
                                       &thrust::raw_reference_cast(*(output_begin + offset)),
                                       output_buffer,
                                       sizeof(value_type) * num_outputs);
#endif  // end CUCO_HAS_CUDA_BARRIER
    }

    if constexpr (not uses_memcpy_async) {
      for (auto index = lane_id; index < num_outputs; index += g.size()) {
        *(output_begin + offset + index) = output_buffer[index];
      }
    }
```